wallet/datastore.go

```diff
@@ -431,47 +357,710 @@ func (pg *Postgres) IncreaseLinkingLimit(ctx context.Context, providerLinkingID
+			Msg("failed to get CustodianLinkCount from DB")
+		return 0, 0, fmt.Errorf("failed to get CustodianLinkCount from DB: %w", err)
+	}
+	// get max
+	stmt = `
+		select ($2 + count(1)) as max from linking_limit_adjust where provider_linking_id = $1
+	`
+	err = tx.Get(&max, stmt, linkingID, getEnvMaxCards())
+	if err != nil {
+		sublogger.Error().Err(err).
+			Msg("failed to get CustodianLinkCount from DB")
+		return 0, 0, fmt.Errorf("failed to get CustodianLinkCount from DB: %w", err)
+	}
+
+	if !noContextTx {
+		// commit
+		err = tx.Commit()
+		if err != nil {
+			sublogger.Error().Err(err).Msg("failed to commit transaction")
+			return 0, 0, fmt.Errorf("failed to commit transaction: %w", err)
+		}
+	}
+
+	return used, max, nil
+}
+
+// GetCustodianLinkByID - get the wallet custodian record by id
+func (pg *Postgres) GetCustodianLinkByID(ctx context.Context, ID uuid.UUID) (*CustodianLink, error) {
+	var (
+		cl  = new(CustodianLink)
+		err error
+	)
+	// create a sublogger
+	sublogger := logger(ctx).With().
+		Str("custodian_link_id", ID.String()).
+		Logger()
+
+	sublogger.Debug().
+		Msg("starting GetCustodianLinkByID")
+
+	// get tx
+	tx, noContextTx := ctx.Value(appctx.DatabaseTransactionCTXKey).(*sqlx.Tx)
+	if !noContextTx {
+		sublogger.Debug().
+			Msg("no tx in context")
+		tx, err = createTx(ctx, pg)
+		if err != nil || tx == nil {
+			sublogger.Error().Err(err).
+				Msg("error creating tx")
+			return nil, fmt.Errorf("failed to create tx for GetCustodianLinkByID: %w", err)
+		}
+		// add tx to ctx for future
+		defer pg.RollbackTx(tx)
+	}
+
+	// query
+	stmt := `
+		select
+			id, wallet_id, custodian, created_at, linked_at, disconnected_at, deposit_destination, linking_id
+		from
+			wallet_custodian
+		where
+			id = $1
+	`
+	err = tx.Get(cl, stmt, ID)
 	if err != nil {
+		sublogger.Error().Err(err).
+			Msg("failed to get CustodianLink from DB")
+		return nil, fmt.Errorf("failed to get CustodianLink from DB: %w", err)
+	}
+
+	if !noContextTx {
+		// commit
+		err = tx.Commit()
+		if err != nil {
+			sublogger.Error().Err(err).Msg("failed to commit transaction")
+			return nil, fmt.Errorf("failed to commit transaction: %w", err)
+		}
+	}
+	return cl, nil
+}
+
+// MigrateCustodianLink - check if this wallet has migrated to multi custodian linkage
+func (pg *Postgres) MigrateCustodianLink(ctx context.Context, walletID uuid.UUID) error {
+	var (
+		wallet walletutils.Info
+		err    error
+	)
+	// create a sublogger
+	sublogger := logger(ctx).With().
+		Str("wallet_id", walletID.String()).
+		Logger()
+
+	sublogger.Debug().
+		Msg("starting MigrateCustodianLink")
+
+	// get tx
+	tx, noContextTx := ctx.Value(appctx.DatabaseTransactionCTXKey).(*sqlx.Tx)
+	if !noContextTx {
+		sublogger.Debug().
+			Msg("no tx in context")
+		tx, err = createTx(ctx, pg)
+		if err != nil || tx == nil {
+			sublogger.Error().Err(err).
+				Msg("error creating tx")
+			return fmt.Errorf("failed to create tx for GetCustodianLinkByID: %w", err)
```

Comment:
comment out of sync

Reply:
resolved: https://github.com/brave-intl/bat-go/pull/793/commits/ca66e903529564cb9ad4e5af367a97fda7fa4f44